internal/control_plane/server/services.go

```diff
@@ -20,7 +22,8 @@ func NewProcServiceServer(exec execution.Execution) protobuf.ClientCPServicesSer
 
 func (s *clientCPServicesServer) PostMetadata(ctx context.Context, request *protobuf.RequestToPostMetadata) (*protobuf.MetadataName, error) {
 	name, err := s.procExec.SaveMetadataToDb(ctx, request.Metadata)
-	logger.Error(err, "Job Create Request Received - Posting Metadata to etcd")
+	uid := id_generator.NextID()
```

Comment:
It will be better if uuid generated is added to the context "ctx" so that we can use it everywhere inside the present execution stack.